src/ert/config/queue_config.py

```diff
@@ -245,88 +247,52 @@ def create_local_copy(self) -> QueueConfig:
 
 
 @no_type_check
 def _check_queue_option_settings(
     selected_queue_system: QueueSystem,
-    queue_options: List[Tuple[str, str]],
+    queue_options: Dict[QueueSystem, Dict[str, str]],
     config_dict: ConfigDict,
     realization_memory: int,
 ) -> None:
-    for queue_system, queue_system_settings in queue_options.items():
-        if queue_system_settings:
+    for queue_system, queue_system_options in queue_options.items():
+        if queue_system_options:
             _validate_queue_driver_settings(
-                queue_system_settings,
+                queue_system_options,
                 realization_memory,
                 QueueSystem(queue_system).name,
                 throw_error=(queue_system == selected_queue_system),
             )
 
-    if (
-        selected_queue_system != QueueSystem.LOCAL
-        and queue_options[selected_queue_system]
-    ):
-        _check_for_overwritten_queue_system_options(
-            selected_queue_system,
-            queue_options[selected_queue_system],
-        )
     if selected_queue_system == QueueSystem.TORQUE:
         _check_num_cpu_requirement(
             config_dict.get("NUM_CPU", 1),
             queue_options[selected_queue_system],
         )
 
 
-def _option_list_to_dict(option_list: List[Tuple[str, str]]) -> Dict[str, List[str]]:
-    temp_dict: Dict[str, List[str]] = defaultdict(list)
-    for option_string in option_list:
-        temp_dict[option_string[0]].append(option_string[1])
-    return temp_dict
-
-
-def _check_for_overwritten_queue_system_options(
-    selected_queue_system: QueueSystem,
-    queue_system_options: List[Tuple[str, str]],
-) -> None:
-    for option_name, option_values in _option_list_to_dict(
-        queue_system_options
-    ).items():
-        if len(option_values) > 1 and option_values[0] != option_values[-1]:
-            logging.info(
-                f"Overwriting QUEUE_OPTION {selected_queue_system} {option_name}:"
-                f" \n Old value: {option_values[0]} \n New value: {option_values[-1]}"
-            )
-
-
 def _check_num_cpu_requirement(
     num_cpu: int,
-    queue_system_options: List[Tuple[str, str]],
+    queue_system_options: Dict[str, str],
 ) -> None:
-    torque_options = _option_list_to_dict(queue_system_options)
-    num_nodes_str = torque_options.get("NUM_NODES", [""])[-1]
-    num_cpus_per_node_str = torque_options.get("NUM_CPUS_PER_NODE", [""])[-1]
-    if not num_nodes_str and not num_cpus_per_node_str:
-        # NUM_CPU will take precedence when deprecated options are not set
+    if (
+        "NUM_NODES" not in queue_system_options
+        and "NUM_CPUS_PER_NODE" not in queue_system_options
+    ):
         return
-    num_nodes = int(num_nodes_str) if num_nodes_str else 1
-    num_cpus_per_node = int(num_cpus_per_node_str) if num_cpus_per_node_str else 1
+    num_nodes = int(queue_system_options.get("NUM_NODES", 1) or "1")
```

Comment:
Do we need the extra `... or "1"` here?

Reply:
Yes, because it is (currently) valid configuriation to supply an empty string for NUM_NODES

Reply:
Why is that valid?

Reply:
Not sure, could be a legacy from the C-driver. We can change it in the future (but separate PR).

Reply:
Maybe all I know is that it will fail current tests if support for empty strings is removed.